src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.CompilationTracker_Generators.cs

```diff
@@ -85,12 +85,13 @@ private partial class CompilationTracker : ICompilationTracker
         }
 
         private async Task<(Compilation compilationWithGeneratedFiles, CompilationTrackerGeneratorInfo generatorInfo)?> TryComputeNewGeneratorInfoInRemoteProcessAsync(
-            SolutionState solution,
+            SolutionCompilationState compilationState,
```

Comment:
So this seems a bit strange: if we're computing new generator results, we don't actually need to sync over the existing CompilationState, right? Is this just because right now we don't have a sync method that takes only the "new" SolutionState? If I recall our conversation we figured this was potentially the one "special" place that can safely ignore the generated state since that's the whole point of this method.

Reply:
I of course am worried if it means we have to fork the sync system to support either kind of state, unless we just can easily introduce some interface for either kind of state if all it needs is a checksum in the end or something.

Reply:
> Is this just because right now we don't have a sync method that takes only the "new" SolutionState? If I recall our conversation we figured this was potentially the one "special" place that can safely ignore the generated state since that's the whole point of this method.

That's exactly right.  And this is something i want to do in a followup.  It's unclear though how best to layer that and use the type system properly.  Especially since *everyone else* will not want this path.

Reply:
> I of course am worried if it means we have to fork the sync system to support either kind of state, unless we just can easily introduce some interface for either kind of state if all it needs is a checksum in the end or something.

Yup.  I don't have a great answer yet.  What i'm tentatively thinking is:

have all the existing  'Invoke' methods (which take the full compilation state).  And then expose the literal: "get me the generator results" APIs which will take only the solution-state.  So basically, no one else would call these.  And this code here would only call that API.